Assets/GothicVR/Scripts/Manager/BarrierManager.cs

```diff
@@ -0,0 +1,138 @@
+using GVR.Caches;
+using GVR.Creator;
+using GVR.Debugging;
+using GVR.Extensions;
+using GVR.Util;
+using UnityEngine;
+
+namespace GVR.GothicVR.Scripts.Manager
+{
+    public class BarrierManager : SingletonBehaviour<BarrierManager>
+    {
+        private GameObject barrier;
+
+        private Material material1;
+        private Material material2;
+        private bool materialsCached;
+
+        private bool barrierVisible;
+
+        private bool barrierFadeIn;
+        private bool fadeIn = true;
+        private bool fadeOut;
+        private float fadeState;
+        private float fadeTime;
+        private float timeUpdatedFade;
+
+        private float nextActivation = 8f;
+
+        private const float BarrierMinOpacity = 0f;
+        private const float BarrierMaxOpacity = 120f;
+
+        // all these values are representing the time in seconds
+        private const float TimeToStayVisible = 25;
+        private const float TimeToStayHidden = 1200;
+        private const float TimeStepToUpdateFade = 0.001f;
+
+
+        public void CreateBarrier()
+        {
+            var barrierMesh = AssetCache.TryGetMsh("MAGICFRONTIER_OUT.MSH");
+            barrier = MeshObjectCreator.CreateBarrier("Barrier", barrierMesh, Vector3.zero, Quaternion.identity)
+                .GetAllDirectChildren()[0];
+        }
+
+        public void FixedUpdate()
```

Comment:
Please add a comment why we need it at FixedUpdate. As you subscribe to gameTimeChange listener, you could also leverage this one to start/stop timed.

Reply:
I wanted a consistent way to check if barrier needs to be faded in/out, and for now the fade in/out state is dependent on how fast we call the function, using fixed update we get a consistent fade in/out